### ISSUE TITLE
Some warnings building 4.5.0 with CMake

### ISSUE THREAD

**Issue by ArchangeGabriel** [Unidata/netcdf-c#501]
### Environment Information

* What platform are you using? (please provide specific distribution/version in summary)
  * [x] Linux
* 32 and/or 64 bit?
  * [ ] 32-bit
  * [x] 64-bit
* What build system are you using?
  * [ ] autotools (`configure`)
  * [x] cmake

Warnings in build log:
```
In file included from /build/netcdf/src/netcdf-c-4.5.0/libdap4/d4includes.h:44:0,
                 from /build/netcdf/src/netcdf-c-4.5.0/libdap4/d4meta.c:6:
/build/netcdf/src/netcdf-c-4.5.0/libdap4/d4meta.c: In function ‘buildCompound’:
/build/netcdf/src/netcdf-c-4.5.0/libdap4/d4meta.c:538:18: warning: passing argument 7 of ‘nc_insert_array_compound’ from incompatible pointer type [-Wincompatible-pointer-types]
            rank, dimsizes)));
                  ^
/build/netcdf/src/netcdf-c-4.5.0/libdap4/ncd4.h:175:32: note: in definition of macro ‘NCCHECK’
 #define NCCHECK(expr) if((ret=(expr))) {ret = NCD4_errorNC(ret,__LINE__,__FILE__); goto done;}else{}
                                ^~~~
In file included from /build/netcdf/src/netcdf-c-4.5.0/libdap4/d4includes.h:33:0,
                 from /build/netcdf/src/netcdf-c-4.5.0/libdap4/d4meta.c:6:
/build/netcdf/src/netcdf-c-4.5.0/include/netcdf.h:593:1: note: expected ‘const int *’ but argument is of type ‘size_t * {aka long unsigned int *}’
 nc_insert_array_compound(int ncid, nc_type xtype, const char *name,
 ^~~~~~~~~~~~~~~~~~~~~~~~
```
```
/build/netcdf/src/netcdf-c-4.5.0/nc_test/util.c: In function ‘inRange’:
/build/netcdf/src/netcdf-c-4.5.0/nc_test/util.c:24:12: warning: statement will never be executed [-Wswitch-unreachable]
     double min = 0.0;
            ^~~
```

**Comment by WardF** [comment]
@DennisHeimbigner Since this is in dap4, any thoughts if this could be related to #500 as @ArchangeGabriel suggests?

**Comment by DennisHeimbigner** [comment]
I will have to check to see what is going on both here and for https://github.com/Unidata/netcdf-c/issues/500

**Comment by DennisHeimbigner** [comment]
The second of these (inRange) is a typo, but should otherwise be ok.
The first (d4meta.c) is a size/type mismatch between dap4 dimensions (size_t)
and netcdf dimensions. On 64bit machine, |size_t| != |int| so there will be a problem.

**Comment by DennisHeimbigner** [comment]
I cannot duplicate the nc_test fail, even on a 64bit machine.
Can you see if you can locate Testing/Temporary/LastTest.log
within the cmake build tree and post it.

**Comment by ArchangeGabriel** [comment]
When I said related to #500, I was thinking the lastest warning in the test itself. ;)

@DennisHeimbigner I’ve posted the log in the #500.

**Comment by DennisHeimbigner** [comment]
Ok, we need to duplicate the error locally. Ward, have you been able to do that?

**Comment by WardF** [comment]
Not as of yet.

**Comment by WardF** [comment]
Fixed by pull request #503
